### PR TITLE
[TD-9] Enlarge the multiselect widgets and sort the options

### PR DESCRIPTION
**Pull request by dpoirier**
First pass - larger widgets, sorted content

### CODE REVIEW

File: tracpro/groups/forms.py [hunk 1]
```diff
@@ -18,6 +22,8 @@ def __init__(self, model, org, *args, **kwargs):
         # Retrieve Contact Group choices from RapidPro.
         choices = [(group.uuid, "%s (%d)" % (group.name, group.size))
                    for group in self.org.get_temba_client().get_groups()]
+        # Sort choices by the labels
+        choices.sort(key=itemgetter(1))
```
Comment (emullaney) [inline tracpro/groups/forms.py]: I'm not exactly sure why but this is only sorting some of the items correctly. A few extra items get appended onto the end of the list:
image

**Comment by dpoirier** [comment]
@emullaney I've changed all the sorts to be case-insensitive.

File: tracpro/baseline/forms.py [hunk 2]
```diff
@@ -32,7 +33,8 @@ def __init__(self, *args, **kwargs):
         super(BaselineTermForm, self).__init__(*args, **kwargs)
 
         if org:
-            polls = Poll.objects.active().by_org(org).order_by('name')
+            polls = Poll.objects.active().by_org(org)\
+                .annotate(lcase_name=Lower('name')).order_by('lcase_name')
```
Comment (emullaney) [inline tracpro/baseline/forms.py]: Can this be simplified as 
`polls = Poll.objects.active().by_org(org).order_by(Lower('name'))`

I don't think annotate is necessary here

File: tracpro/baseline/forms.py [hunk 3]
```diff
@@ -62,20 +64,22 @@ def label_from_instance(self, obj):
 class SpoofDataForm(forms.Form):
     """ Form to create spoofed poll data """
     contacts = forms.ModelMultipleChoiceField(
-        queryset=Contact.objects.all(),
-        help_text=_("Select contacts for this set of spoofed data."))
+        queryset=Contact.objects.annotate(lcase_name=Lower('name')).order_by('lcase_name'),
```
Comment (emullaney) [inline tracpro/baseline/forms.py]: See previous comment RE: annotate

File: tracpro/baseline/forms.py [hunk 4]
```diff
@@ -62,20 +64,22 @@ def label_from_instance(self, obj):
...
+        widget=forms.widgets.SelectMultiple(attrs={'size': '20'}),
+    )
     start_date = forms.DateField(
         help_text=_("Baseline poll data will be submitted on this date. "
                     "Follow up data will start on this date."))
     end_date = forms.DateField(
         help_text=_("Follow up data will end on this date. "))
     baseline_question = QuestionModelChoiceField(
-        queryset=Question.objects.all().order_by('poll__name', 'text'),
+        queryset=Question.objects.annotate(lcase_poll__name=Lower('poll__name')).order_by('lcase_poll__name', 'text'),
```
Comment (emullaney) [inline tracpro/baseline/forms.py]: annotate change

File: tracpro/baseline/forms.py [hunk 5]
```diff
@@ -62,20 +64,22 @@ def label_from_instance(self, obj):
...
         help_text=_("Select a baseline question which will have numeric "
                     "answers only."))
     follow_up_question = QuestionModelChoiceField(
         label=_("Follow Up Question"),
-        queryset=Question.objects.all().order_by('poll__name', 'text'),
+        queryset=Question.objects.annotate(lcase_poll__name=Lower('poll__name')).order_by('lcase_poll__name', 'text'),
```
Comment (emullaney) [inline tracpro/baseline/forms.py]: annotate change

File: tracpro/baseline/forms.py [hunk 6]
```diff
@@ -99,7 +103,7 @@ def __init__(self, *args, **kwargs):
         super(SpoofDataForm, self).__init__(*args, **kwargs)
 
         if org:
-            contacts = Contact.objects.active().by_org(org).order_by('name')
+            contacts = Contact.objects.active().by_org(org).annotate(lcase_name=Lower('name')).order_by('lcase_name')
```
Comment (emullaney) [inline tracpro/baseline/forms.py]: annotate change

File: tracpro/baseline/forms.py [hunk 7]
```diff
@@ -169,7 +173,7 @@ def __init__(self, baseline_term, data_regions, *args, **kwargs):
         else:
             queryset = data_regions
         queryset = queryset.filter(pk__in=baseline_term.get_regions())
-        queryset = queryset.order_by('name')
+        queryset = queryset.annotate(lcase_name=Lower('name')).order_by('lcase_name')
```
Comment (emullaney) [inline tracpro/baseline/forms.py]: annotate change

File: tracpro/contacts/forms.py [hunk 8]
```diff
@@ -35,9 +36,11 @@ def __init__(self, *args, **kwargs):
         self.fields['name'].required = True
         self.fields['group'].required = True
 
-        self.fields['region'].queryset = self.user.get_all_regions(org)
+        regions = self.user.get_all_regions(org)\
+            .annotate(lcase_name=Lower('name')).order_by('lcase_name')
```
Comment (emullaney) [inline tracpro/contacts/forms.py]: annotate change

File: tracpro/contacts/forms.py [hunk 9]
```diff
@@ -35,9 +36,11 @@ def __init__(self, *args, **kwargs):
...
+        self.fields['region'].queryset = regions
         self.fields['group'].empty_label = ""
-        self.fields['group'].queryset = Group.get_all(org).order_by('name')
+        self.fields['group'].queryset = Group.get_all(org).annotate(lcase_name=Lower('name')).order_by('lcase_name')
```
Comment (emullaney) [inline tracpro/contacts/forms.py]: annotate change

File: tracpro/groups/views.py [hunk 10]
```diff
@@ -130,7 +131,8 @@ def derive_queryset(self, **kwargs):
             return regions
 
         def get_context_data(self, **kwargs):
-            org_boundaries = Boundary.objects.by_org(self.request.org).order_by('name')
+            org_boundaries = Boundary.objects.by_org(self.request.org)\
+                .annotate(lcase_name=Lower('name')).order_by('lcase_name')
```
Comment (emullaney) [inline tracpro/groups/views.py]: annotate change

File: tracpro/home/views.py [hunk 11]
```diff
@@ -23,7 +24,7 @@ def has_permission(self, request, *args, **kwargs):
 
     def get_context_data(self, **kwargs):
         polls = Poll.objects.active().by_org(self.request.org)
-        polls = polls.order_by('name')
+        polls = polls.annotate(lcase_name=Lower('name')).order_by('lcase_name')
```
Comment (emullaney) [inline tracpro/home/views.py]: annotate change

File: tracpro/orgs_ext/forms.py [hunk 12]
```diff
@@ -48,6 +78,11 @@ def __init__(self, *args, **kwargs):
         self.fields['available_languages'].initial = self.instance.available_languages or []
         self.fields['show_spoof_data'].initial = self.instance.show_spoof_data or False
 
+        # Sort the administrators
+        self.fields['administrators'].queryset \
+            = self.fields['administrators'].queryset\
+            .annotate(lcase_full_name=Lower('profile__full_name')).order_by('lcase_full_name')
```
Comment (emullaney) [inline tracpro/orgs_ext/forms.py]: annotate comment

File: tracpro/orgs_ext/forms.py [hunk 13]
```diff
@@ -66,7 +101,8 @@ def __init__(self, *args, **kwargs):
                     raise
 
             data_fields = self.instance.datafield_set.all()
-            self.fields['contact_fields'].queryset = data_fields
+            self.fields['contact_fields'].queryset = data_fields\
+                .annotate(lcase_label=Lower('label')).order_by('lcase_label')
```
Comment (emullaney) [inline tracpro/orgs_ext/forms.py]: annotate comment

File: tracpro/polls/charts.py [hunk 14]
```diff
@@ -168,7 +169,9 @@ def multiple_pollruns_numeric_split(pollruns, answers, responses, question, cont
     sum_data = []
     avg_data = []
     rate_data = []
-    for region in Region.objects.filter(pk__in=data.keys()).order_by('name'):
+    regions = (Region.objects.filter(pk__in=data.keys())
+               .annotate(lcase_name=Lower('name')).order_by('lcase_name'))
```
Comment (emullaney) [inline tracpro/polls/charts.py]: annotate change

File: tracpro/polls/forms.py [hunk 15]
```diff
@@ -46,7 +49,8 @@ def __init__(self, org, *args, **kwargs):
         # NOTE: This makes an in-band request to an external API.
         models.Poll.objects.sync(self.org)
 
-        polls = models.Poll.objects.by_org(self.org)
+        polls = models.Poll.objects.by_org(self.org)\
+            .annotate(lcase_rapidpro_name=Lower('rapidpro_name')).order_by('lcase_rapidpro_name')
```
Comment (emullaney) [inline tracpro/polls/forms.py]: annotate change

File: tracpro/profiles/forms.py [hunk 16]
```diff
@@ -41,10 +42,12 @@ class UserForm(forms.ModelForm):
         help_text=_("Whether user must change password on next login."))
     regions = ModifiedLevelTreeNodeMultipleChoiceField(
         label=_("Regions"),
-        queryset=Region.objects.all(),
+        queryset=Region.objects.annotate(lcase_name=Lower('name')).order_by('lcase_name'),
```
Comment (emullaney) [inline tracpro/profiles/forms.py]: annotate change

**Comment by dpoirier** [comment]
That's so much simpler!   I've updated the pull request.

**Comment by emullaney** [comment]
:cat2:

**Pull request merged**